erpnext/manufacturing/doctype/work_order/work_order.py

```diff
@@ -212,6 +212,7 @@ def update_work_order_qty(self):
 					self.meta.get_label(fieldname), qty, completed_qty, self.name), StockOverProductionError)
 
 			self.db_set(fieldname, qty)
+			frappe.db.set_value('Sales Order Item', self.sales_order_item, 'produced_qty', self.produced_qty)
```

Comment:
There can be multiple work order against same sales order item. In that case, it should consider sum(produced_qty)